src/services/patternMatcher.ts

```diff
@@ -0,0 +1,813 @@
+        function getMatches(candidate: string, dottedContainer: string): PatternMatch[] {
+            if (skipMatch(candidate)) {
+                return undefined;
+            }
+
+            // First, check that the last part of the dot separated pattern matches the name of the
+            // candidate.  If not, then there's no point in proceeding and doing the more
+            // expensive work.
+            var candidateMatch = matchSegment(candidate, lastOrUndefined(dotSeparatedSegments));
+            if (!candidateMatch) {
+                return undefined;
+            }
+
+            dottedContainer = dottedContainer || "";
+            var containerParts = dottedContainer.split(".");
+
+            // -1 because the last part was checked against the name, and only the rest
+            // of the parts are checked against the container.
+            if (dotSeparatedSegments.length - 1 > containerParts.length) {
+                // There weren't enough container parts to match against the pattern parts.
+                // So this definitely doesn't match.
+                return null;
+            }
+
+            // So far so good.  Now break up the container for the candidate and check if all
+            // the dotted parts match up correctly.
+            var totalMatch = candidateMatch;
+
+            for (var i = dotSeparatedSegments.length - 2, j = containerParts.length - 1;
+                 i >= 0;
+                 i--, j--) {
+
+                var segment = dotSeparatedSegments[i];
+                var containerName = containerParts[j];
+
+                var containerMatch = matchSegment(containerName, segment);
+                if (!containerMatch) {
+                    // This container didn't match the pattern piece.  So there's no match at all.
+                    return undefined;
+                }
+
+                addRange(totalMatch, containerMatch);
+            }
+
+            // Success, this symbol's full name matched against the dotted name the user was asking
+            // about.
+            return totalMatch;
+        }
+
+        function getWordSpans(word: string): TextSpan[] {
+            if (!hasProperty(stringToWordSpans, word)) {
+                stringToWordSpans[word] = breakIntoWordSpans(word);
+            }
+
+            return stringToWordSpans[word];
+        }
+
+        function matchTextChunk(candidate: string, chunk: TextChunk, punctuationStripped: boolean): PatternMatch {
+            var index = indexOfIgnoringCase(candidate, chunk.textLowerCase);
+            if (index === 0) {
+                if (chunk.text.length === candidate.length) {
+                    // a) Check if the part matches the candidate entirely, in an case insensitive or
+                    //    sensitive manner.  If it does, return that there was an exact match.
+                    return createPatternMatch(PatternMatchKind.Exact, punctuationStripped, /*isCaseSensitive:*/ candidate === chunk.text);

```

Comment:
Your comment implies that there is something to check, but you are just creating the match off the bat. Where did you actually check that it matched?